### PR TITLE
Virtualbox ubuntu packages

### PR DESCRIPTION
**Pull request by paskino**
Currently the UPDATE.sh is not called at creation time as I wanted to test a few details of the VM I created before building SIRF.

### CODE REVIEW

File: scripts/UPDATE.sh [hunk 1]
```diff
@@ -231,6 +231,16 @@ fi
 # copy scripts into the path
 cp -vp $SIRF_SRC_PATH/CCPPETMR_VM/scripts/update*sh $SIRF_INSTALL_PATH/bin
 
+# install the SIRF-Exercises
+cd $SIRF_SRC_PATH
+clone_or_pull SIRF-Exercises
+# install ipython notebook and firefox
```
Comment (KrisThielemans) [inline scripts/UPDATE.sh]: 2 comment lines to be deleted

File: scripts/UPDATE.sh [hunk 2]
```diff
@@ -231,6 +231,16 @@ fi
...
+#$SIRF_SRC_PATH/CCPPETMR_VM/scripts/update_VM_with_exercises.sh
+echo "#! /bin/bash
```
Comment (KrisThielemans) [inline scripts/UPDATE.sh]: these ones cannot be here.
something like this should be in the SIRF-exercises/scripts repo.
Presumably we need different content for the jupyter notebook. (if you like, could have 2 scripts, but really that's overkill for now).

File: vagrant/Vagrantfile [hunk 3]
```diff
@@ -152,10 +158,15 @@ Vagrant.configure("2") do |config|
     cd $HOME/devel
 	
     git clone https://github.com/CCPPETMR/CCPPETMR_VM.git
-	cd CCPPETMR_VM
-	git pull
-	git checkout install_new_packages
+    cd CCPPETMR_VM
+    git pull
+    git checkout install_new_packages
```
Comment (KrisThielemans) [inline vagrant/Vagrantfile]: this will have to go of course

File: vagrant/Vagrantfile [hunk 4]
```diff
@@ -152,10 +158,15 @@ Vagrant.configure("2") do |config|
     cd $HOME/devel
 	
     git clone https://github.com/CCPPETMR/CCPPETMR_VM.git
-	cd CCPPETMR_VM
-	git pull
-	git checkout install_new_packages
+    cd CCPPETMR_VM
+    git pull
```
Comment (KrisThielemans) [inline vagrant/Vagrantfile]: by the way, not necessary to do a pull just after a clone

File: scripts/INSTALL_prerequisites_with_apt-get.sh [hunk 5]
```diff
@@ -21,10 +21,7 @@ $SUDO apt-get install -y --no-install-recommends swig
 
 # TODO replace with pip install
 echo "Installing python libraries etc"
-$SUDO apt-get install -y --no-install-recommends  python-scipy python-docopt  python-numpy python-h5py python-matplotlib python-libxml2 python-psutil spyder python-tk python-nose
+$SUDO apt-get install -y --no-install-recommends  python-scipy python-docopt  python-numpy python-h5py python-matplotlib python-libxml2 python-psutil python-tk python-nose
```
Comment (KrisThielemans) [inline scripts/INSTALL_prerequisites_with_apt-get.sh]: please add python-pip here

File: scripts/update_VM_with_exercises.sh [hunk 6]
```diff
@@ -0,0 +1,6 @@
+#! /bin/bash
+# script to install in the VM the packages needed for training.
+# Originally introduced for the PSMR2018 training event.
+
+echo "installing ipython notebook and browser"
```
Comment (KrisThielemans) [inline scripts/update_VM_with_exercises.sh]: message is incorrect. In fact, currently this file is not necessary as it's all elsewhere

File: vagrant/Vagrantfile [hunk 7]
```diff
@@ -107,6 +107,12 @@ Vagrant.configure("2") do |config|
 	apt-mark hold grub-pc
 	# upgrade the list of packages
 	apt-get upgrade -y
+	# uninstall the ubuntu provided virtualbox packages
+	# https://github.com/CCPPETMR/CCPPETMR_VM/issues/77#issuecomment-382686304
+	apt-get install build-essential module-assistant
+	sudo /opt/VBoxGuestAdditions-*/uninstall.sh
+        apt-get purge virtualb*
```
Comment (KrisThielemans) [inline vagrant/Vagrantfile]: I think this has to be
```
apt-get remove --purge virtualb*
```

**Comment by KrisThielemans** [comment]
I see you've added `python-qt4`. I'm somewhat worried about mixing apt-get and pip, but then I discovered that pip2 doesn't seem to have pyqt, so we'd have to switch to python3. aargh. obviously not for now.

File: scripts/UPDATE.sh (hunk 2 above)
Comment (paskino) [reply scripts/UPDATE.sh]: What cannot be here? The `start_notebook`?

File: scripts/INSTALL_prerequisites_with_apt-get.sh (hunk 5 above)
Comment (paskino) [reply scripts/INSTALL_prerequisites_with_apt-get.sh]: I also add firefox

File: scripts/update_VM_with_exercises.sh (hunk 6 above)
Comment (paskino) [reply scripts/update_VM_with_exercises.sh]: I will remove this. I was unsure where all this should stay. I initially created it with the idea to call it only if one wants to create a VM with jupyter and the exercises.

File: scripts/INSTALL_prerequisites_with_apt-get.sh (hunk 5 above)
Comment (KrisThielemans) [reply scripts/INSTALL_prerequisites_with_apt-get.sh]: yes, but not in this file, but in Vagrantfile, which seemed fine to us. could leave the pip in Vagrantfile as well if you prefer actually.

File: scripts/UPDATE.sh (hunk 2 above)
Comment (paskino) [reply scripts/UPDATE.sh]: On Slack @KrisThielemans  and I have agreed that we will add info in the SIRF-Exercises repository on how to start the jupyter notebook.

File: scripts/INSTALL_prerequisites_with_apt-get.sh (hunk 5 above)
Comment (paskino) [reply scripts/INSTALL_prerequisites_with_apt-get.sh]: Could you tell me your rationale? So maybe I will do it right, next time.

Comment (paskino) [reply scripts/INSTALL_prerequisites_with_apt-get.sh]: Actually, reading the initial comment it seems to describe the rationale. However, as far as I understand, @bathomas is not exactly using  `INSTALL_prerequisites_with_apt-get.sh` and `UPDATE.sh`. He has a [`provision.sh`](https://github.com/bathomas/stir-azure/blob/master/provision.sh) which does a mixture.

**Comment by paskino** [comment]
I just want to resume:

1. The creation of VM and Azure should do only the minimal amount of operations differently
2. VM uses `INSTALL_prerequisites_with_apt-get.sh` to install prerequisites
3. VM uses `UPDATE.sh` to install SIRF

Azure aligns by using the 2. and 3. However, there are significant differences:
1. VM is created from `Vagrantfile`
2. Azure is created from [`provision.sh`](https://github.com/bathomas/stir-azure/blob/master/provision.sh)

Specific commands for each should stay in either file. In particular, both require installing jupyter, but VM requires additional install: `firefox`, `spyder` and `python-qt4` on the VM. 

Therefore I suggest to have 
 *. `python -m pip install jupyter` in the `INSTALL_prerequisites_with_apt-get.sh`
 *. `firefox`, `spyder` and `python-qt4` in the Vagrantfile

**Comment by KrisThielemans** [comment]
Agreed with the strategy. 

As it turns out, @bathomas 's `provision.sh` now does use `INSTALL_prerequisites_with_apt-get.sh`, [check here](https://github.com/bathomas/stir-azure/blob/d7407e81448c63407e42a5be743bd8b5e1da71db/provision.sh#L34). we've discussed this more, and he will cut out most of the python stuff in his file as it overlaps with `INSTALL_prerequisites_with_apt-get.sh` (probably only needing to install `git`). So, `provision.sh` will become quite short (great!).

I still wouldn't put `python -m pip install jupyter` in the `INSTALL_prerequisites_with_apt-get.sh`. For one thing, the file has the wrong name :-;

anyway. all of this is related to https://github.com/CCPPETMR/SIRF-SuperBuild/issues/91 and https://github.com/CCPPETMR/SIRF-SuperBuild/issues/92

**Comment by KrisThielemans** [comment]
looks good to me. would have to remove the branch checkout before merging of course.

can we rename this to vb.name = "SIRF_1.1.0-rc.1-test" or something?

**Comment by paskino** [comment]
Double checking: should I install jupyter in the Vagrantfile, just because `INSTALL_prerequisite_with_apt-get.sh` bears the *wrong* name?

**Comment by KrisThielemans** [comment]
seems a trivial reason maybe, but to me it avoids confusion. Also, Ben might be using conda for this and not pip.

**Comment by paskino** [comment]
It's not trivial. Would you consider adding 
1. `INSTALL_prerequisites.sh` which will call `INSTALL_prerequisites_with_apt-get.sh` and then `pip install`, or
2. `INSTALL_prerequisites_with_pip.sh`
?

**Comment by KrisThielemans** [comment]
We could, but now is not the time. (i know, i should have said that yesterday as well, but i was optimistic!).

**Pull request merged**